tremor-script/src/grammar.lalrpop

```diff
@@ -23,7 +24,172 @@ use beef::Cow;
+    <start:@L> "define" "pipeline" <id:Ident> <query:EmbeddedQuery> <end:@L> => DeployStmtRaw::PipelineDecl(PipelineDeclRaw { start, end, id: id.to_string(), query, args: vec![], params: None }),
+    // Connectors - these define connectors for deployment
+    <start:@L> "define" <kind:ConnectorKind> "connector" <id:Ident> <args:ArgSpecificationClause> "end" <end:@L> => DeployStmtRaw::ConnectorDecl(ConnectorDeclRaw { start, end, id: id.to_string(), kind, args: args.0, params: args.1 }),
+    <start:@L> "define" <kind:ConnectorKind> "connector" <id:Ident> "end" <end:@L> => DeployStmtRaw::ConnectorDecl(ConnectorDeclRaw { start, end, id: id.to_string(), kind, args: vec![], params: None }),
+    // Flow - these define how pipelines and connectors are interconnected to form a unit deployment graph
+    <start:@L> "define" "flow" <id:Ident> <args:ArgSpecificationClause> <links:WithDeployLinks> "end" <end:@L> => DeployStmtRaw::FlowDecl(FlowDeclRaw { start, end, id: id.to_string(), args: args.0, params: args.1, links }),
+    <start:@L> "define" "flow" <id:Ident> <links:WithDeployLinks> "end" <end:@L> => DeployStmtRaw::FlowDecl(FlowDeclRaw { start, end, id: id.to_string(), args: vec![], params: None, links }),
+
+    <start:@L> "deploy" <id:Ident> "from" <target:ModularTarget> <args:ArgFinalizationClause> <end:@L> => DeployStmtRaw::CreateStmt(CreateStmtRaw { start, end, id, module: target.0, target: target.1, params: args }),
+    <start:@L> "deploy" <id:Ident> "from" <target:ModularTarget> <end:@L> => DeployStmtRaw::CreateStmt(CreateStmtRaw { start, end, id: id.clone(),  module: target.0, target: target.1, params: None }),
+    <start:@L> "deploy" <id:Ident> <args:ArgFinalizationClause> <end:@L> => DeployStmtRaw::CreateStmt(CreateStmtRaw { start, end, id: id.clone(),  module: vec![], target: id, params: args }),
+    <start:@L> "deploy" <id:Ident>  <end:@L> => DeployStmtRaw::CreateStmt(CreateStmtRaw { start, end, id: id.clone(), module: vec![], target: id, params: None }),
+}
+
+// Allows arguments to be specified and defaults to be provided and overridden in definitions
+ArgSpecificationClause: (WithArgsRaw<'input>, Option<DeployWithExprsRaw<'input>>) = {
+    <args:DeployArgs> <params:DeployConfigClause> => (args, Some(params)),
+    <args:DeployArgs> => (args, None),
+    <params:DeployConfigClause> => (vec![], Some(params)),
+}
+
+// Allows arguments to be consumed with defaults or overriden in deployment statements
+ArgFinalizationClause: Option<DeployWithExprsRaw<'input>> = {
+    <params:DeployWithClause> "end" => Some(params),
+}
+
+ConnectorKind: (Vec<IdentRaw<'input>>, IdentRaw<'input>) = {
+    <mt:ModularTarget> => <>,
+}
+
+DeployArgs: WithArgsRaw<'input> = {
+    "args" <with:WithArgs> => <>,
+}
+
+DeployModuleStmt: DeployModuleStmtRaw<'input> = {
```

Comment:
I think we need to split the deployment statements into defines and deploys, having defines in modules follows what what tremor script (does today) and trickle (will do with Yatins pr) but we have always excluded anything non definitional in modules and it seems sensible to do the same here.

```troy
mod a with
 define flow snot #...
end
```
should be legal, but 


```troy
mod a with
 define flow snot #...
 deploy flow snot;
end
```
should not

Reply:
ignore this, I got confused in .larlpop naming :D this is already the case

Reply:
`deploy` is illegal in module statements ( module statement inner production doesn't allow it )
`deploy` is currently constrained to the top level in troy - we'd need extensive changes to the rfc
and further design work to consider more modular deployments. But modular definitions are fine and
consistent with what we already allow in the query language, and Yatin is extending for sub queries.

Also - i think we want the sub-query work merged before considering modular deployments - for example
we likely want to make constraints on pipeline definitions in troy that are fairly strict ( no definitions ) in the
troy production. Yatin likely has good insights here based on his work on modular subquery here as well.